link/uprobe_test.go

```diff
@@ -248,3 +249,44 @@ func TestUprobePathOffset(t *testing.T) {
 		})
 	}
 }
+
+func TestUprobeProgramCall(t *testing.T) {
+	// Create ebpf map. Will contain only one key with initial value 0.
+	m, err := ebpf.NewMap(&ebpf.MapSpec{
+		Type:       ebpf.Array,
+		KeySize:    4,
+		ValueSize:  4,
+		MaxEntries: 1,
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Create ebpf program. When called, will set the value of key 0 in
+	// the map created above to 1.
+	p := newMapUpdaterProg(t, m, ebpf.Kprobe)
+
+	// Open Uprobe on '/bin/bash' for the symbol 'main'
+	// and attach it to the ebpf program created above.
+	u, err := bashEx.Uprobe(bashSym, p)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer func(l Link) {
+		if err := l.Close(); err != nil {
+			t.Fatal(err)
+		}
+	}(u)
+
+	// Trigger ebpf program call.
+	_ = exec.Command("/bin/bash", "42").Run()
```

Comment:
I think this should check the return code, since the exec itself may fail if /bin/bash doesn't exist:
```suggestion
	err := exec.Command("/bin/bash", "-c", "true").Run()
	if err != nil { ... }
```

Reply:
Follow up thought: use `/bin/sh` instead of bash? That will have main as well.

Reply:
using `/bin/sh` works for me locally, CI is complaining though https://ebpf.semaphoreci.com/jobs/6efeb5d2-b89e-44ac-ac90-3f331cefbdc0

Reply:
Ok, let's stick to bin/bash then. Thanks for giving it a shot!